### PR TITLE
App 823 - Minting Summary

### PR DESCRIPTION
**Pull request by WissenIstNacht**
## Description

On the token minting action of proposal, adds the necessary logic to display:
- count of additional tokens after the mint
- count of additional token holders after the mint
- count of new total token supply after the mint

It does NOT include the count of new total token after the mint, as the necessary currently does not exist on testnets.

Task: [710](https://aragonassociation.atlassian.net/browse/APP-710)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: packages/web-app/src/containers/actionBuilder/mintTokens/index.tsx [hunk 1]
```diff
@@ -1,22 +1,55 @@
...
 import {AccordionMethod} from 'components/accordionMethod';
+import {useActionsContext} from 'context/actions';
+import {useNetwork} from 'context/network';
+import {useProviders} from 'context/providers';
+import {CHAIN_METADATA} from 'utils/constants';
+import {fetchBalance, getTokenInfo} from 'utils/tokens';
 import {AddressAndTokenRow} from './addressTokenRow';
+import {BigNumber} from 'ethers';
+import {useDaoParam} from 'hooks/useDaoParam';
+import {useDaoToken} from 'hooks/useDAoToken';
```
Comment (sepehr2github) [inline packages/web-app/src/containers/actionBuilder/mintTokens/index.tsx]: Small Typo ;D
```suggestion
import {useDaoToken} from 'hooks/useDaoToken';
```

File: packages/web-app/src/containers/actionBuilder/mintTokens/index.tsx [hunk 2]
```diff
@@ -25,6 +58,77 @@ const MintTokens: React.FC<Props> = ({index}) => {
...
+      }
+    }
+  }, [mints, daoToken.id]);
+
+  useEffect(() => {
+    // Collecting token amounts that are to be minted
+    if (mints && daoToken) {
+      let newTokensCount = 0;
+      mints.forEach(m => {
+        newTokensCount += parseInt(m.amount);
```
Comment (sepehr2github) [inline packages/web-app/src/containers/actionBuilder/mintTokens/index.tsx]: amount is BigInt, You need to calculate decimals first then add the new amount to corresponding number!

File: packages/web-app/src/containers/actionBuilder/mintTokens/index.tsx [hunk 3]
```diff
@@ -25,6 +58,77 @@ const MintTokens: React.FC<Props> = ({index}) => {
...
+              //add address to promise to keep track later
+              return {address: m.address, balance: b};
+            })
+        );
+        Promise.all(promises).then((abs: AddressBalance[]) => {
+          // new holders are addresses that have 0 balance for token
+          const holderAddresses = abs.filter((ab: AddressBalance) =>
+            ab.balance.isZero()
+          );
+          setNewTokenHolders(prev => {
```
Comment (sepehr2github) [inline packages/web-app/src/containers/actionBuilder/mintTokens/index.tsx]: When I entered my wallet address, It added +1 to new holders. but then when removed the address the calculation didn't change!

File: packages/web-app/src/containers/actionBuilder/mintTokens/index.tsx [hunk 4]
```diff
@@ -120,26 +229,28 @@ const MintTokens: React.FC<Props> = ({index}) => {
...
+              </p>
+            </HStack>
+            <HStack>
+              <Label>{t('labels.newHolders')}</Label>
+              <p>+{newTokenHolders.size}</p>
+            </HStack>
+            <HStack>
+              <Label>{t('labels.totalTokens')}</Label>
+              <p>
+                {totalTokens.toString()} {daoToken.symbol}
```
Comment (sepehr2github) [inline packages/web-app/src/containers/actionBuilder/mintTokens/index.tsx]: Screen Shot 2022-08-04 at 11 37 50 AM

When you go to the next page and then come back the total Token will be reset!

**Review by sepehr2github — CHANGES_REQUESTED** [review]
Thanks, Vali. Seems good, left a few comments.

Question: the token minting form validations weren't part of the ticket, right? If yes, do we have any tickets regarding the corresponding topic?

**Review by RakeshUP — APPROVED** [review]

**Review by sepehr2github — APPROVED** [review]
Thanks for the addressing comments, Just 1 small point to remind for later. Once the user changes the token amount, the address should be required! We can fix this later once we want to add the validations.
Screen Shot 2022-08-08 at 12 14 42 PM

**Pull request merged**